opencti-platform/opencti-front/src/private/components/common/stix_nested_ref_relationships/StixNestedRefRelationshipCreationFromEntityLine.tsx

```diff
@@ -0,0 +1,445 @@
+  selectAll: boolean;
+  onToggleShiftEntity: (
+    index: number,
+    entity: StixNestedRefRelationshipCreationFromEntityLine_node$data,
+    event?: React.SyntheticEvent
+  ) => void;
+  index: number;
+}
+
+export const StixNestedRefRelationshipCreationFromEntityLine: FunctionComponent<StixNestedRefRelationshipCreationFromEntityLineProps> = ({
+  node,
+  dataColumns,
+  onLabelClick,
+  onToggleEntity,
+  selectedElements,
+  deSelectedElements,
+  selectAll,
+  onToggleShiftEntity,
+  index,
+}) => {
+  const classes = useStyles();
+  const { t_i18n } = useFormatter();
+  const data = useFragment(stixNestedRefRelationshipCreationFromEntityLineFragment, node);
+  const flag = data.entity_type === 'Country'
+    && R.head((data.x_opencti_aliases ?? []).filter((n) => n?.length === 2));
+  return (
+    <ListItem
+      classes={{ root: classes.item }}
+      divider={true}
+      button={true}
+      onClick={(event) => (event.shiftKey
+        ? onToggleShiftEntity(index, data, event)
+        : onToggleEntity(data, event))
+      }
+    >
+      <ListItemIcon
+        classes={{ root: classes.itemIcon }}
+        style={{ minWidth: 40 }}
+        onClick={(event) => (event.shiftKey
+          ? onToggleShiftEntity(index, data, event)
+          : onToggleEntity(data, event))
+        }
+      >
+        <Checkbox
+          edge="start"
+          checked={
+            (selectAll && !(data.id in (deSelectedElements || {})))
+            || data.id in (selectedElements || {})
+          }
+          disableRipple={true}
+        />
+      </ListItemIcon>
+      <ListItemIcon classes={{ root: classes.itemIcon }}>
+        {flag ? (
+          <img
+            style={{ width: 20 }}
+            src={`${APP_BASE_PATH}/static/flags/4x3/${flag.toLowerCase()}.svg`}
+            alt={data.name}
+          />
+        ) : (
+          <ItemIcon type={data.entity_type} />
+        )}
+      </ListItemIcon>
+      <ListItemText
+        primary={
+          <div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.entity_type.width }}
+            >
+              <Chip
+                classes={{ root: classes.chipInList }}
+                style={{
+                  backgroundColor: hexToRGB(itemColor(data.entity_type), 0.08),
+                  color: itemColor(data.entity_type),
+                  border: `1px solid ${itemColor(data.entity_type)}`,
+                }}
+                label={t_i18n(`entity_${data.entity_type}`)}
+              />
+            </div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.value.width }}
+            >
+              {defaultValue(data)}
+            </div>
+            <div
+              className={classes.bodyItem}
+              style={{ width: dataColumns.createdBy.width }}
+            >
+              {R.pathOr('', ['createdBy', 'name'], data)}
```

Comment:
idem, data['createdBy']['name'] ?

Reply:
ramda does some magic if the path, at some point, leads to undefined.

so it's equivalent to `data.?createdBy?.name ?? ''` in modern js